include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -4413,6 +4418,20 @@ struct OrtApi {
    * \since Version 1.16.
    */
   ORT_API2_STATUS(KernelContext_GetResource, _In_ const OrtKernelContext* context, _In_ int resouce_version, _In_ int resource_id, _Outptr_ void** resource);
+
+  /** \brief Set user logging function
```

Comment:
Need to document how this interacts with the env logger, and what kind of output we can expect. Does it override the environment contained logger?